components/form-control/script.js

```diff
@@ -238,11 +352,108 @@ class GamefaceFormControl extends HTMLElement {
         }
     }
 
+    /**
+     * Checks the validator method
+     * @param {{method:()=>boolean}} validator
+     * @returns {boolean}
+     */
+    hasValidMethodProperty(validator) {
+        return validator !== undefined && validator.method && typeof validator.method === 'function';
+    }
+
+    /**
+     * Checks the validator error message
+     * @param {{errorMessage:(element:HTMLElement)=>string}} validator
+     * @returns {boolean}
+     */
+    hasValidMessageProperty(validator) {
+        return validator !== undefined && validator.errorMessage && typeof validator.errorMessage === 'function';
+    }
+
+    /**
+     * Will return the validator object by the error type
+     * @param {{[errorType: string]: {method?: Function<boolean>, errorMessage?: (element:HTMLElement) => string}}} elementCustomValidators 
```

Comment:
It will be valid in case of the existance of the `@typedef`s.
```suggestion
     * @param {Object<Validator>} elementCustomValidators
```

Reply:
You mean `{Object<string,Validator>}` I made a different definition that is named 'Validators'